extensions/ql-vscode/src/commandRunner.ts

```diff
@@ -42,6 +42,11 @@ export interface ProgressUpdate {
 
 export type ProgressCallback = (p: ProgressUpdate) => void;
 
+// Make certain properties within a type optional
+type Optional<T, K extends keyof T> = Pick<Partial<T>, K> & Omit<T, K>;
+
+export type ProgressOptions = Optional<VSCodeProgressOptions, "location">;
```

Comment:
This is cool, but I'm wondering what the advantage is over having a union of two types: one with the location attribute and one without?

Reply:
I can see you want to use `VSCodeProgressOptions`, but it is harder to read than two types added together. 

🤔 Just wondering how to make this easier to understand when you first look at it.

Reply:
I didn't want to copy over the types of `VSCodeProgressOptions`, but you're right that this makes it harder to read. We can copy over the types of that type, but that means potentially some additional work when we update the `vscode` types, which might not be too bad for this type. What do you think?

Reply:
Yeah, `ProgressOptions` and `ProgressLocation` don't look that elaborate so it wouldn't be that much work to change. But we would need to remember to do it, which we probably won't unless we have solid testing around what we expect the progress UI to look like. 

Since I don't think we have testing at this level, I'm leaning towards just using the root types like you said.